pgmoon/socket.moon

```diff
@@ -20,10 +20,8 @@ luasocket = do
       sock = @sock
       original = sock[key]
       if type(original) == "function"
-        fn = (_, ...) ->
+        (_, ...) ->
```

Comment:
I'm worried about performance here. Every time we call a socket method we now are hitting the metatable lookup function (much slower than table access) and creating a new anonymous function. A single query probably does this ~5 times. 

I see two solutions:

1. purge the function cache when we know we need to point to a new socket by just setting everything to `nil`
2. don't create a closure on a reference to the socket when the function was first created, but access it in `@` on every call (not familiar with what luasec is actually doing, so maybe this might not work)